### PR TITLE
[NeuralChat] Refine askdoc API and retrieval logic

### PR DESCRIPTION
**Pull request by letonghan**
## Type of Change

feature
API not changed.

## Description

Refine askdoc backend API, modify display format of links, and refine retieval logic.

## Expected Behavior & Potential Risk

The quality of askdoc example will be improved.

## How has this PR been tested?

local tested.

## Dependency Change?

no.

### CODE REVIEW

File: intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/retrieval_agent.py [hunk 1]
```diff
@@ -88,7 +88,7 @@ def append_localdb(self,
                        append_path, 
                        top_k=1, 
                        search_type="similarity_score_threshold", 
-                       search_kwargs={"score_threshold": 0.9, "k": 1}):
+                       search_kwargs={"score_threshold": 0.8, "k": 1}):
```
Comment (kevinintel) [inline intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/retrieval_agent.py:91]: why 0.8?

File: intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/detector/intent_detection.py [hunk 2]
```diff
@@ -16,10 +16,17 @@
 # limitations under the License.
 """Check the intent of the input user query with LLM."""
 
-from intel_extension_for_transformers.neural_chat.pipeline.plugins.prompt.prompt_template \
-    import generate_intent_prompt
+# from intel_extension_for_transformers.neural_chat.pipeline.plugins.prompt.prompt_template \
```
Comment (hshen14) [inline intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/detector/intent_detection.py:19]: why not using the existing generate_intent_prompt?

Comment (letonghan) [reply intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/detector/intent_detection.py:19]: It's for better performance of retrieval, maybe @XuhuiRen could explain this better than me.

File: intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/retrieval_agent.py (hunk 1 above)
Comment (letonghan) [reply intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/retrieval_agent.py:91]: Same like another comment, this is also for higher quality of retrieval

File: intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/detector/intent_detection.py (hunk 2 above)
Comment (XuhuiRen) [reply intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/detector/intent_detection.py:19]: > It's for better performance of retrieval, maybe @XuhuiRen could explain this better than me.

It should follow the previous usage. I ask you to put it here just for the convenience of testing the askgm demo. I will fix it.

Comment (letonghan) [reply intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/detector/intent_detection.py]: okay, I misunderstood.

**Review by lvliang-intel — APPROVED** [review]

**Pull request merged**